_adr/88/index.adoc

```diff
@@ -1,48 +1,71 @@
 ---
 num: 88
-title: Authorization for control plane admin APIs
-status: "Draft" # One of Draft, Accepted, Rejected
+title: Authorization for fleet-manager admin APIs
+status: "Draft"
 authors:
-  - "akoserwal"
+  - "ppaszki"
 tags:
-  - "sso" # e.g. kafka, connectors, registry
+  - "sso"
 applies_padrs: # What PADRs does this ADR apply?
 applies_patterns: # What APs does this ADR apply?
 ---
 
-// Top style tips:
-// * Use one sentence per line
-// * No unexpanded acronyms
-// * No undefined jargon
-
-// No need for a title heading, it's added by the template
-
 ## Context and Problem Statement
-// What is the background against which this decision is being taken?
+The purpose of this ADR is to define a standardized approach for securing admin API in a fleet manager
 
 ## Goals
-// Bulleted list of outcomes that this ADR, if accepted, should help achieve
+* Provide standardized way to secure admin API endpoints for fleet managers
 
 ## Non-goals
-// Bulleted list of outcomes that this ADR is not trying to achieve.
 
 ## Current situation
-// Where are we now?
 
 ## Proposal
-// What is the decision being proposed
+
+image::adr-88-admin-api-architecture-diagram.png[architecture diagram]
+
+A Keycloak realm is used to authorize access to admin API endpoint(s). 
+Such realm contains roles that allow access to endpoints to the privileged API users.
+Recommended approach would be to create a set of realm roles per environment and name them in the way that indicates the permissions they grant as follows:
+
+* <unique-fleet-manager-prefix>-admin-read-stage
+* <unique-fleet-manager-prefix>-admin-write-stage
+* <unique-fleet-manager-prefix>-admin-full-stage
+
+* <unique-fleet-manager-prefix>-admin-read-prod
+* <unique-fleet-manager-prefix>-admin-write-prod
+* <unique-fleet-manager-prefix>-admin-full-prod
+
+These roles map to the following permissions for the environment included as the role postfix (e.g. `prod`):
+
+* \*-read-* - list permissions
+* \*-write-* - list/ update permissions
+* \*-full-* - full permissions (list/ update/ delete)
+
+Example full role for kas-fleet-manager in production environment is: `kas-fleet-manager-admin-full-prod`.
+
+Once Keycloak instance is set up, a privileged user uses their credentials to get a JWT token from Keycloak.
+The JWT token is sent to a admin API in a HTTP request.
+Admin API authorization middleware verifies token signature with Public JWKS Certificate and roles included in the JWT token are checked against roles specified in the fleet manager.
+Matching role for given admin endpoint grants access to the action associated with that endpoint.
```

Comment:
Can we cross-reference with the sequence numbers in the diagram?

Reply:
done